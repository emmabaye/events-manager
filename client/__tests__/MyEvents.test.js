import { BrowserRouter } from 'react-router-dom';
import ConnectedMyEvents, { MyEvents } from '../components/MyEvents.jsx';

describe('MyEvents Component', () => {
  const initialState = {
    myEvents: {
      data: {
        rows: []
      }
    },
    eventReducer: {
      userEvents: {
        data: {
          rows: [],
          page: {}
        }
      }
    },
  };


  const store = mockStore(initialState);

  global.localStorage = {
    getItem: (str) => str
  };

  describe('MyEvents form  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = mount(<BrowserRouter><MyEvents /></BrowserRouter>);
      expect(wrapper).toMatchSnapshot();
    });

    it('it should render component', () => {
      const wrapper = mount(<BrowserRouter><Provider store={store}><ConnectedMyEvents /></Provider></BrowserRouter>);
      expect(wrapper.length).toEqual(1);
    });
  });
});
