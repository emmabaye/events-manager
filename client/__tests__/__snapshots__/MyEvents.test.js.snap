// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyEvents Component MyEvents form  should render self it should render for dumb component 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MyEvents />
  </Router>
</BrowserRouter>
`;
